### ISSUE TITLE
Loss of text selection when opening right-click menu

### ISSUE THREAD

**Issue by Angelk90** [zed-industries/zed#14008]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Zed:

https://github.com/zed-industries/zed/assets/20476002/30ddd4c9-6d36-4500-ac0f-22ccdb9aa284

Vscode:

https://github.com/zed-industries/zed/assets/20476002/f83ca3cc-1139-4ce6-8471-773e6b30983d

### Environment

```
Zed: v0.143.4 (Zed Preview)
OS: macOS 14.1.1
Memory: 16 GiB
Architecture: x86_64
```

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by notpeter** [comment]
I am able to reproduce this issue with Zed 0.143.6 (current stable) but not with Zed 0.144.0 (current preview). Can you confirm you no longer see this in your Zed Preview after updating?

**Comment by notpeter** [comment]
This has been fixed in preview (v0.143.1) and will be available in stable on Wednesday 2024-07-17.
- https://github.com/zed-industries/zed/pull/13701
- https://github.com/zed-industries/zed/pull/14150

Thanks for reporting!
